### ISSUE TITLE
Run error

### ISSUE THREAD

**Issue by Lanskih90** [blockchain/service-my-wallet-v3#13]
Hi, 
I'm trying to run the blockchain wallet service, command: 

>blockchain-wallet-service start --port 3000

Result error:

>[ReferenceError: navigator is not defined]
[ReferenceError: window is not defined]
Error Decrypting Wallet. Please check your password is correct

What's the problem?
Thanks!

**Comment by JakePrice86** [comment]
I am getting this issue also.

Every time I try to use the login method (localhost:3000/merchant/GUID/login) I get the error Error Decrypting Wallet. Please check your password is correct. I have tried with my accounts, and have even setup another account with a more simple password (one without symbols) and still cannot get on. IP address is whitelisted as well. When checking with the logging in the user panel, I can see this for every time I try the login:

Today 15:10:06    viewed login page    IP Address    Unknown

Tried everything, but nothing works!

**Comment by kristovatlas** [comment]
Did you follow these directions when requesting an API key? 
> Get an API code here. Note: You must check the "Create Wallets" checkbox under "Permissions" when requesting an API code in order for it to be compatible with this app.

**Comment by JakePrice86** [comment]
I have an API key - two in fact. Both still show the same error.

**Comment by KMHouk** [comment]
What system are you running? Windows/Linux/Mac?

These two messages are normal:
>[ReferenceError: navigator is not defined]
>[ReferenceError: window is not defined]


Did you get any errors during installation? Can you try a complete npm uninstall & reinstall? 

A common error is not having Java JVM, which during installation would say something like:
> To use the Java command line tool you need to install a jdk

**Comment by Lanskih90** [comment]
I use Linux Ubuntu 14, instaled JDK java v. 1.7.0_91 !
During installation there were errors:
https://github.com/blockchain/service-my-wallet-v3/issues/12

**Comment by Lanskih90** [comment]
npm version 2.14.14

**Comment by JakePrice86** [comment]
Ubuntu 14.04
JDK installed
npm version 3.5.2
node version 4.2.4

npm uninstall and install done. Still getting same error.

**Comment by mladen-stankovic** [comment]
I'm getting the exact same error:
"Error Decrypting Wallet. Please check your password is correct"
but only on Windows!
I have Java, Visual Studio 2013 Express and every node-gyp prerequisite installed .
I get no error when installing the wallet service.

On the other hand, I've installed wallet service on Debian Linux and everything works

**Comment by leifdroms** [comment]
Can anyone tell me how to unsubscribe from this email chain?

Thanks
Leif

**Comment by Lanskih90** [comment]
Someone tell me how to solve my problem?)

**Comment by Lanskih90** [comment]
How to remove service-my-wallet-v3 from the server? 
I want try to reinstall.

**Comment by Lanskih90** [comment]
I reinstalled all programs!!!!

I have run blockchain-wallet-service (3000 port), but now, when make request the on the address http://localhost:3000/merchant/:guid/login?password=pass&api_code=code , 
I have error:

>[ReferenceError: navigator is not defined]
[ReferenceError: window is not defined]
ERR_HISTORY

Doing the query in **php** using **file_get_contents($url);** return **(false)**

**What is the API !!! The second day I sit can't run it!!! At each step the error!!!**

**Comment by JakePrice86** [comment]
My issue is caused by the package:

    "blockchain-wallet-client": "^3.3.1"

It doesn't seem to decrypt the wallet properly - it always fails on this line (wallet-crypto.js line 167)

    var decoded = decrypted.toString(CryptoJS.enc.Utf8);

To get round the issue, I have skipped the decrypting and manually added my unencrypted wallet to the file like this (from line 211) :

```
try {

      /*
      decryptedWallet = decryptAes(jsonWrapper.payload, password, jsonWrapper.pbkdf2_iterations);
      jsonWallet = JSON.parse(decryptedWallet);
      */
      jsonWallet = JSON.parse('DECRYPTED WALLET HERE');

    } catch (e) {
```

Hope this helps out someone before the deadline in a day!

**Comment by jtormey** [comment]
@Lanskih90 I noticed you said you were using node v0.10, can you try upgrading to at least v0.12 and seeing if that fixes it? Also, with the latest version of the service, you should get a more helpful error message than `ERR_HISTORY` if it still fails.

@JakePrice86 is it an old wallet you are using? I'm looking into what might be causing the decryption error.

Note: we have extended the deadline for migrating from Wallet API v1, as the first deadline was unreasonable. The new deprecation date is about two months from now.

If anyone has specific questions about the API, feel free to contact me directly.

**Comment by KMHouk** [comment]
@JakePrice86 It is not recommended that you do this, because it will interfere with wallet syncing. 

I just made a fresh Ubuntu box 14.04 and going from 0 to fully deploy, the commands were:

1. Install git `apt-get install git`
2. Install Java's JDK `apt-get install default-jdk`
3. Installed g++ `apt-get install g++`
4. Installed service from GitHub `npm install -g git+https://github.com/blockchain/service-my-wallet-v3.git`

Can you uninstall the service and follow the steps above to see if it works for you?

**Comment by JakePrice86** [comment]
Perfect @KMHouk 

The only issues I seem to get now are random errors with stream writing back the user - which I have got round with forever.

```
/var/www/chain/node_modules/readable-stream/lib/_stream_writable.js:202
    encoding = state.defaultEncoding;
                    ^

TypeError: Cannot read property 'defaultEncoding' of undefined
    at DuplexWrapper.Writable.write (/var/www/chain/node_modules/readable-stream/lib/_stream_writable.js:202:21)
    at DuplexWrapper.Writable.end (/var/www/chain/node_modules/readable-stream/lib/_stream_writable.js:426:10)
    at API.request (/var/www/chain/node_modules/blockchain-wallet-client/src/api.js:91:30)
    at API.retry (/var/www/chain/node_modules/blockchain-wallet-client/src/api.js:118:12)
    at API.securePost (/var/www/chain/node_modules/blockchain-wallet-client/src/api.js:239:15)
    at Object.MyWallet.securePost (/var/www/chain/node_modules/blockchain-wallet-client/src/wallet.js:64:7)
    at Object.MyWallet.getWallet (/var/www/chain/node_modules/blockchain-wallet-client/src/wallet.js:745:12)
    at WebSocket.onMessage (/var/www/chain/node_modules/blockchain-wallet-client/src/wallet.js:221:18)
    at emitTwo (events.js:87:13)
    at WebSocket.emit (events.js:172:7)
    at Receiver.ontext (/var/www/chain/node_modules/ws/lib/WebSocket.js:816:10)
    at /var/www/chain/node_modules/ws/lib/Receiver.js:477:18
    at /var/www/chain/node_modules/ws/lib/Receiver.js:361:7
    at /var/www/chain/node_modules/ws/lib/PerMessageDeflate.js:238:5
    at afterWrite (_stream_writable.js:346:3)
    at onwrite (_stream_writable.js:337:7)

**Comment by jtormey** [comment]
Glad the decryption is working! Can you check in your node_modules/ to see which version of blockchain-wallet-client you have? If it is lower than v3.3.3, try upgrading it. The error you're seeing was supposedly fixed in the last version.

**Comment by JakePrice86** [comment]
I was on v3.3.1 - upgraded to v3.3.3 but now get:

```
/var/www/chain/node_modules/readable-stream/lib/_stream_writable.js:207
  if (state.ended)
           ^

TypeError: Cannot read property 'ended' of undefined
    at DuplexWrapper.Writable.write (/var/www/chain/node_modules/readable-stream/lib/_stream_writable.js:207:12)
    at DuplexWrapper.Writable.end (/var/www/chain/node_modules/readable-stream/lib/_stream_writable.js:426:10)
    at API.request (/var/www/chain/node_modules/blockchain-wallet-client/src/api.js:91:30)
    at API.retry (/var/www/chain/node_modules/blockchain-wallet-client/src/api.js:118:12)
    at API.securePost (/var/www/chain/node_modules/blockchain-wallet-client/src/api.js:239:15)
    at Object.MyWallet.securePost (/var/www/chain/node_modules/blockchain-wallet-client/src/wallet.js:64:7)
    at Object.MyWallet.getWallet (/var/www/chain/node_modules/blockchain-wallet-client/src/wallet.js:745:12)
    at WebSocket.onMessage (/var/www/chain/node_modules/blockchain-wallet-client/src/wallet.js:221:18)
    at emitTwo (events.js:87:13)
    at WebSocket.emit (events.js:172:7)
    at Receiver.ontext (/var/www/chain/node_modules/ws/lib/WebSocket.js:816:10)
    at /var/www/chain/node_modules/ws/lib/Receiver.js:477:18
    at /var/www/chain/node_modules/ws/lib/Receiver.js:361:7
    at /var/www/chain/node_modules/ws/lib/PerMessageDeflate.js:238:5
    at afterWrite (_stream_writable.js:346:3)
    at onwrite (_stream_writable.js:337:7)
[nodemon] app crashed - waiting for file changes before starting...

**Comment by Lanskih90** [comment]
@jtormey - I've updated nodejs to v0.12 , now I can not start the service, error :

> $ sudo blockchain-wallet-service start --port 3000
/usr/local/lib/node_modules/service-my-wallet-v3/node_modules/bcrypt/node_modules/bindings/bindings.js:83
        throw e
              ^
Error: Module did not self-register.
    at Error (native)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at bindings (/usr/local/lib/node_modules/service-my-wallet-v3/node_modules/bcrypt/node_modules/bindings/bindings.js:76:44)
    at Object.<anonymous> (/usr/local/lib/node_modules/service-my-wallet-v3/node_modules/bcrypt/bcrypt.js:3:35)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/usr/local/lib/node_modules/service-my-wallet-v3/src/api.js:6:15)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)

**Comment by jtormey** [comment]
@Lanskih90 try rebuilding your dependencies: `npm rebuild`. Or, if that doesn't work, run a fresh install: `rm -rf node_modules && npm install`.

**Comment by Lanskih90** [comment]
@jtormey `npm rebuild` - doesn't work, and `rm -rf node_modules && npm install` error:

>npm ERR! install Couldn't read dependencies
npm ERR! Linux 3.13.0-57-generic
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "install"
npm ERR! node v0.12.9
npm ERR! npm  v2.14.9
npm ERR! path /home/user/package.json
npm ERR! code ENOPACKAGEJSON
npm ERR! errno -2

>npm ERR! package.json ENOENT, open '/home/bit-apple/package.json'
npm ERR! package.json This is most likely not a problem with npm itself.
npm ERR! package.json npm can't find a package.json file in your current directory.

>npm ERR! Please include the following file with any support request:
npm ERR!     /home/user/npm-debug.log

**Comment by jtormey** [comment]
Sorry, I should have been more specific about how to do that. You need to be in the same directory as blockchain-wallet-service in order to rebuild it. Here is a step-by-step:

```sh
realpath $(which blockchain-wallet-service) # find the absolute path of the wallet service
cd "$(dirname $(!!))/.." # change to the directory of the wallet service
npm rebuild # rebuild it
```

Hope that helps.

**Comment by jtormey** [comment]
`Cannot read property 'X' of undefined` errors fixed as of v0.10.1
